src/interactivity/onecore/ConIoSrvComm.hpp

```diff
@@ -38,25 +38,27 @@ namespace Microsoft::Console::Interactivity::OneCore
 
         [[nodiscard]] NTSTATUS RequestGetDisplaySize(_Inout_ PCD_IO_DISPLAY_SIZE pCdDisplaySize) const;
         [[nodiscard]] NTSTATUS RequestGetFontSize(_Inout_ PCD_IO_FONT_SIZE pCdFontSize) const;
-        [[nodiscard]] NTSTATUS RequestSetCursor(_In_ CD_IO_CURSOR_INFORMATION* const pCdCursorInformation) const;
+        [[nodiscard]] NTSTATUS RequestSetCursor(_In_ const CD_IO_CURSOR_INFORMATION* const pCdCursorInformation) const;
         [[nodiscard]] NTSTATUS RequestUpdateDisplay(_In_ SHORT RowIndex) const;
 
-        [[nodiscard]] USHORT GetDisplayMode() const;
+        [[nodiscard]] USHORT GetDisplayMode() const noexcept;
 
-        PVOID GetSharedViewBase() const;
+        PVOID GetSharedViewBase() const noexcept;
 
         VOID CleanupForHeadless(const NTSTATUS status);
 
         [[nodiscard]] NTSTATUS InitializeBgfx();
         [[nodiscard]] NTSTATUS InitializeWddmCon();
 
-        Microsoft::Console::Render::WddmConEngine* pWddmConEngine;
+        Render::WddmConEngine wddmConEngine;
 
     private:
         [[nodiscard]] NTSTATUS EnsureConnection();
         [[nodiscard]] NTSTATUS SendRequestReceiveReply(PCIS_MSG Message) const;
 
-        VOID HandleFocusEvent(PCIS_EVENT const FocusEvent);
+        VOID HandleFocusEvent(const CIS_EVENT* const FocusEvent);
+
+        Render::BgfxEngine _bgfxEngine;

```

Comment:
Why allocate the `WddmConEngine` / `BgfxEngine` on the heap when you can just allocate it on the stack? heh.
No manual new/delete = no AuditMode complaints.

Reply:
I'm worried about giving these things default constructors since they represent or require access to unique platform objects... and only one of them will ever be active at a time. I'd rather keep them on the heap (not that it's expensive to allocate them on the heap after all) than have to worry about what it means for two of them to technically exist simultaneously. Thoughts?